tutorials/arvo/parsing.md

```diff
@@ -24,37 +24,38 @@ Parsing a string is rarely done all at once. Instead, it is usually done
 character-by-character, and the return contains the data structure representing
 what has been parsed thus far as well as the remainder of the string to be
 parsed. They also need to be able to fail in case the input is improperly
-formed. We will see each of these standard practices implemented in Hoon below.
+formed. We will see each of these standard practices implemented in [Hoon below](#parsing-in-hoon).
 
 ## Functional parsers
 
 How parsers are built varies substantially depending on what sort of programming
 language it is written in. As Hoon is a functional programming language, we will
 be focused on understanding _functional parsers_, also known as _combinator
-parsers_.
+parsers_. In this section we will make light use of pseudocode, as introducing
+the Hoon to describe functional parsers here creates a chicken-and-egg problem.
 
 Complex functional parsers are built piece by piece from simpler parsers
 that are plugged into one another in various ways to perform the desired task.
 
 The basic building blocks, or primitives, are parsers that read only a
 single character. There are frequently a few types of possible input characters,
-such as letters, numbers, and symbols. For example, `parse(integer, "1")` calls
+such as letters, numbers, and symbols. For example, `parse("1", integer)` calls
 the parsing routine on the string `"1"` and looks for an integer, and so it
 returns the integer `1`. However, taking into account what was said above about
 parsers returning the unparsed portion of the string as well, we should
 represent this return as a tuple. So we should expect something like this:
 ```
-> parse(integer, "1")
+> parse("1", integer)
 (1, "")
-> parse(integer, "123")
+> parse("123", integer)
 (1, "23")
```

Comment:
I suggested changing the pseudo-code to look more like hoon (ie `(parse "1" integer)`, `[1 ""]` etc). You gave the comment a :+1:. Is the idea here that you don't want to get it confused for something you might try running?

Reply:
I thought you were just saying that I should mention that it is pseudocode. I think it definitely makes sense to change the pseudocode to be more Hoon-like though.